src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -730,9 +742,41 @@ private void closeTempFile(String encodedPartition) {
   }
 
   private void closeTempFile() {
+    ConnectException connectException = null;
     for (String encodedPartition : tempFiles.keySet()) {
       // Close the file and propagate any errors
-      closeTempFile(encodedPartition);
+      try {
+        closeTempFile(encodedPartition);
+      } catch (ConnectException e) {
+        // still want to close all of the other data writers
+        connectException = e;
+        log.error(
+            "Failed to close temporary file for partition " + encodedPartition
```

Comment:
nit: let's use `{}` for strings like we do elsewhere for consistency. In practice this message won't be skipped.

Reply:
oops nice catch!